### PR TITLE
Added a SplitView sample app for guidance on how to use Prism with the Navigation Menu pattern

### PR DESCRIPTION
**Pull request by ali-hk**
Using Prism with the Navigation Menu pattern via SplitView isn't as straightforward as just a single frame app, so I think this sample will help developers understand how they're supposed to set up the app when using a non-standard shell such as when using a SplitView to host a navigable content frame. Since the Navigation Menu pattern is a common theme in system UWP apps, it's gained wider interest for Windows 10 app in general as well and so it was a good candidate for a sample of a non-standard app shell.

### CODE REVIEW

**Comment by ali-hk** [comment]
Note that there's a TODO to automatically update the menu items based on what page the app is currently on, but this is blocked on the recent changes to Prism.Windows that add navigation state change events (the code is checked in, but the NuGet package hasn't been updated).

**Comment by briannoyes** [comment]
So a couple of questions/comments about this sample.

What purpose is the Frame in the menu pane serving? No navigation happens within it, so why would you not just put your ItemsControl or the MenuPage itself as the content of the SplitView.Pane?

I typically would not call something a ViewModel unless it has a corresponding View. So I would call MenuItemViewModel something more like MenuItemInfo (just a data structure that supports a ViewModel, but not one itself) and derive from BindableBase. ViewModelBase has methods related to navigation that will never apply to something that is just being used as a collection of data items to render into an ItemsControl. 

If you could let me know your thoughts on those, then I can decide what changes to make in pulling this into the repo.

Thanks
Brian

**Comment by ali-hk** [comment]
Good questions :)
Regarding the Frame in the MenuPage, the intent was to keep the ViewModel code for the menu separate and also allow the ViewModle AutoWire to do it's work. You could theoretically do the menu in the Shell itself and have the menu ViewModel code be in a Shell ViewModel instead. It also provides the opportunity of having a multi-level menu.
Admittedly this is a bit heavy for the common usage scenario though, so I could just simplify it and move the menu's ViewModel to a ShellViewModel instead.

Regarding the naming of ViewModels, I followed the pattern from AdventureWorks.Shopper where all ViewModels are named ViewModel (ex. [ProductViewModel.cs](https://github.com/PrismLibrary/Prism-Samples-Windows/blob/master/AdventureWorks.Shopper/AdventureWorks.UILogic/ViewModels/ProductViewModel.cs) ). I think this makes a lot of sense, because names like MenuItemInfo are names that I'd consider more appropriate for Model classes.
Having said that, you're right, it probably doesn't need to be inheriting from ViewModelBase and should probably just be inheriting from BindableBase.

Let me know what you think and I can make the necessary changes.

**Comment by briannoyes** [comment]
I would still keep the menu and its view model as a separate View/ViewModel pair, but would implement the menu view as a UserControl (as MenuView), not a page, and then statically place them into the SplitView.Pane as a child element. The View can still use ViewModelLocator to hook up to its ViewModel, there is no requirement for the view to be a page or be in a Frame for VML to work.

So I would get rid of the Frame in the SplitView.Pane, change MenuPage to MenuView (as a user control), and just add an instance of MenuView to the split view pane as its content element. 

The renaming part is a preference I have, but you are correct that there are other places in our samples where there are "ViewModels" that don't have a corresponding view, so no need to rename those.

Thanks
Brian

**Comment by ali-hk** [comment]
Makes sense, not sure why I didn't think of using a UserControl :) I'll make that change and update this thread when it's done.

**Comment by ali-hk** [comment]
Updated to use a UserControl, but there's a caveat. The UserControl can't be in the AppShell.xaml file declaratively because the INavigationService and IResourceLoader haven't yet been added to the container, which means when it tries to resolve the MenuViewModel, it fails. 
Of course, there isn't a FrameNavigationService until InitializeFrameAsync executes CreateNavigationService, which obviously requires the actual content Frame first.
In short there's a bit of a chicken and egg problem here. My current solution is to expose the SplitView.Pane as a property off of AppShell and set it to a new MenuView in code (in OnLaunchApplicationAsync).

I'll think about how this problem can be addressed better going forward, possibly with a change in the way PrismApplication works.

**Comment by briannoyes** [comment]
Let me take a look. We should be able to construct the shell through the container and expect the services to already be registered by the time we do that. If not, I'll look at moving some of that registration earlier in the lifecycle.

**Comment by briannoyes** [comment]
OK, got it working as intended. The issue was that it was creating the shell in the OnCreateRootFrame override, which is too early in the lifecycle for doing that. The chicken-egg problem is that the frame has to be created and passed to the navigation service instance on creation before it is available from the container. So the solution is to create the shell in the CreateShell override and take the frame that is passed to CreateShell and place it into the shell where desired.

**Comment by ali-hk** [comment]
Ah, I see, that makes sense now :) Makes sense why CreateShell is taking the frame as a parameter. 
It's a bit unintuitive and unclear when going based off the comments what the intent is with the overrides. Hence this sample :)
Thanks!
